Code-4-Community/speak-for-the-trees-frontend
# Leaderboard

## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/702429655/views/12631042/pulses/816228893)

What benefit does this bring to the end user? Or, what benefit does this bring to developers working in the codebase?

On the team and user leaderboard pages, this leaderboard allows users to see a ranked list of teams or users. There is customizability with the number of items per page, the number of tabs, the functionality of the right side of an item, and the collapsible content

## This PR

Describe the changes required and any implementation choices you made to give context to reviewers.

The leaderboard can be seen at the /dummy route. The size and spacing will be changed within the context of where the leaderboard is used so right now there is no styling on its width.

I added a few items to the antd global styling. As you can see in the screenshot, the coloring on the pagination items is not quite right. I am trying to avoid overriding antd styles and I have not been able to figure out the name of that specific item. I changed the primary color to lightgreen because all of the text colors were defaulting to blue.

## Screenshots

Provide screenshots of any new components, styling changes, or pages.

## Verification Steps

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves.

I have a couple of examples of the kinds of collapse panels in the leaderboard, made the dummy page to check that interactions were correct. Current usage only has one tab.

![No collapse](https://user-images.githubusercontent.com/19194912/102140945-e3f9d080-3e14-11eb-84df-349f5168a5d6.PNG)

![with collapse](https://user-images.githubusercontent.com/19194912/102140953-e8be8480-3e14-11eb-97e3-8dea68cd2be8.PNG)

![page 2](https://user-images.githubusercontent.com/19194912/102140964-f07e2900-3e14-11eb-920b-a0a383cd8a54.PNG)
